src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java

```diff
@@ -95,29 +105,46 @@ public void doBuildWithParameters(StaplerRequest req, StaplerResponse rsp, @Quer
         		values.add(value);
         	}
         }
-        Jenkins.getInstance().getQueue().schedule(p, p.getDelay(req), new ParametersAction(values), getBuildCause(req));
+        Jenkins.getInstance().getQueue().schedule(p, delay.getTime(), new ParametersAction(values), getBuildCause(req));
         ok(rsp);
     }
 
     public void doPolling(StaplerRequest req, StaplerResponse rsp, @QueryParameter String job) throws IOException, ServletException {
         LOGGER.log(Level.FINE, "polling on {0}", job);
-        project(job, req, rsp).schedulePolling();
+        ParameterizedJobMixIn.ParameterizedJob p = project(job, req, rsp);
+        // AbstractProject.schedulePolling only adds one thing here: check for isDisabled. But in that case, !isBuildable, so we would not have gotten here anyway.
+        SCMTriggerItem scmp = SCMTriggerItem.SCMTriggerItems.asSCMTriggerItem(p);
+        if (scmp == null) {
+            LOGGER.log(Level.FINE, "{0} is not a SCMTriggerItem", p);
+            throw HttpResponses.error(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, new IOException(job + " is not a SCMTriggerItem"));
```

Comment:
I think you are right, a 400 would be more appropriate. The HTTP spec is kind of vague about what it means.